incremental.go

```diff
@@ -0,0 +1,159 @@
+package feedx
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/bsm/bfs"
+)
+
+// IncrmentalProduceFunc returns a ProduceFunc closure around an incremental mod time
+type IncrementalProduceFunc func(time.Time) ProduceFunc
+
+// IncrementalProducer produces a continuous incremental feed.
+type incrementalProducer struct {
+	*producerState
+
+	bucket    bfs.Bucket
+	manifest  *bfs.Object
+	ownBucket bool
+	opt       incrementalProducerOptions
+
+	ctx  context.Context
+	stop context.CancelFunc
+	ipfn IncrementalProduceFunc
+}
+
+type incrementalProducerOptions ProducerOptions
```

Comment:
Pls keep in mind that this is unsustainable. the options for an incremental producer will be different, since we need 2 intervals (one for produce, one for compaction cycle). I start to think that your original approach with two separate producer types is better as it gives us more flexibility. Maybe try DRYing it up a bit